integrations/access/opsgenie/app.go

```diff
@@ -17,15 +17,575 @@ limitations under the License.
+	data, err := a.opsgenie.CreateAlert(ctx, reqID, reqData)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	ctx, log := logger.WithField(ctx, "opsgenie_alert_id", data.AlertID)
+	log.Info("Successfully created Opsgenie alert")
+
+	// Save opsgenie alert info in plugin data.
+	_, err = a.modifyPluginData(ctx, reqID, func(existing *PluginData) (PluginData, bool) {
+		var pluginData PluginData
+		if existing != nil {
+			pluginData = *existing
+		} else {
+			// It must be impossible but lets handle it just in case.
+			pluginData = PluginData{RequestData: reqData}
+		}
+		pluginData.OpsgenieData = data
+		return pluginData, true
+	})
+	return trace.Wrap(err)
+}
+
+// postReviewNotes posts alert notes about new reviews appeared for request.
+func (a *App) postReviewNotes(ctx context.Context, reqID string, reqReviews []types.AccessReview) error {
+	var oldCount int
+	var data OpsgenieData
+
+	// Increase the review counter in plugin data.
+	ok, err := a.modifyPluginData(ctx, reqID, func(existing *PluginData) (PluginData, bool) {
+		if existing == nil {
+			return PluginData{}, false
+		}
+
+		if data = existing.OpsgenieData; data.AlertID == "" {
+			return PluginData{}, false
+		}
+
+		count := len(reqReviews)
+		if oldCount = existing.ReviewsCount; oldCount >= count {
+			return PluginData{}, false
+		}
+		pluginData := *existing
+		pluginData.ReviewsCount = count
+		return pluginData, true
+	})
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	if !ok {
+		logger.Get(ctx).Debug("Failed to post the note: plugin data is missing")
+		return nil
+	}
+	ctx, _ = logger.WithField(ctx, "opsgneie_alert_id", data.AlertID)
+
+	slice := reqReviews[oldCount:]
+	if len(slice) == 0 {
+		return nil
+	}
+
+	errors := make([]error, 0, len(slice))
+	for _, review := range slice {
+		if err := a.opsgenie.PostReviewNote(ctx, data.AlertID, review); err != nil {
+			errors = append(errors, err)
+		}
+	}
+	return trace.NewAggregate(errors...)
+}
+
+// tryApproveRequest attempts to submit an approval if the requesting user is on-call in one of the services provided in request annotation.
+func (a *App) tryApproveRequest(ctx context.Context, req types.AccessRequest) error {
+	log := logger.Get(ctx)
+
+	serviceNames, err := a.getOnCallServiceNames(req)
+	if err != nil {
+		logger.Get(ctx).Debugf("Skipping the approval: %s", err)
+		return nil
+	}
+
+	onCallUsers := []string{}
+	for _, scheduleName := range serviceNames {
+		respondersResult, err := a.opsgenie.GetOnCall(ctx, scheduleName)
+		if err != nil {
+			return trace.Wrap(err)
+		}
+		onCallUsers = append(onCallUsers, respondersResult.Data.OnCallRecipients...)
+	}
+
+	userIsOnCall := false
+	for _, user := range onCallUsers {
+		if req.GetUser() == user {
+			userIsOnCall = true
+		}
+	}
+	if userIsOnCall {
+		if _, err := a.teleport.SubmitAccessReview(ctx, types.AccessReviewSubmission{
+			RequestID: req.GetName(),
+			Review: types.AccessReview{
+				Author:        req.GetUser(),
```

Comment:
This is the Teleport user that made the request, isn't it? If so, this is probably not the user you want to have issue the review as well - they will most likely not have permissions to do so. 

I added the "@teleport-access-approval-bot" (aka `teleport.SystemAccessApproverUserName`) user as part of the PagerDuty plugin, who as permissions to approve _any_ access request, which is probably the account you want here.